scripts/src/javatests/org/oppia/android/scripts/common/BazelClientTest.kt

```diff
@@ -331,28 +331,34 @@ class BazelClientTest {
       manifestName = "AndroidManifest.xml",
       dependencyName = "//third_party:com_android_support_support-annotations"
     )
-    tempFolder.newFolder("third_party")
-    val thirdPartyBuild = tempFolder.newFile("third_party/BUILD.bazel")
-    thirdPartyBuild.writeText(
-      """
-      load("@rules_jvm_external//:defs.bzl", "artifact")
-      android_library(
-          name = "com_android_support_support-annotations",
-          visibility = ["//visibility:public"],
-          exports = [
-              artifact("com.android.support:support-annotations:28.0.0")
-          ],
-      )
-      """.trimIndent() + "\n"
-    )
-
+    createAndroidLibraryAtThirdPartyBuild("com.android.support:support-annotations:28.0.0")
     val bazelClient = BazelClient(tempFolder.root)
     val thirdPartyDependenciesList =
       bazelClient.retrieveThirdPartyMavenDepsListForBinary("//:test_oppia")
     assertThat("@maven//:com_android_support_support_annotations")
       .isIn(thirdPartyDependenciesList)
   }
 
+  @Test
+  fun testRetrieveMavenDepsList_binaryDependsOnArtifactNotViaThirdParty_doesNotreturnArtifact() {
+    testBazelWorkspace.initEmptyWorkspace()
+    testBazelWorkspace.setupWorkspaceForRulesJvmExternal(
+      listOf("com.android.support:support-annotations:28.0.0")
+    )
+    tempFolder.newFile("AndroidManifest.xml")
+    createAndroidBinary(
+      binaryName = "test_oppia",
+      manifestName = "AndroidManifest.xml",
+      dependencyName = ":com_android_support_support-annotations"
+    )
+    createAndroidLibraryAtRootBuild("io.fabric.sdk.android:fabric:1.4.7")
+    val bazelClient = BazelClient(tempFolder.root)
+    val thirdPartyDependenciesList =
+      bazelClient.retrieveThirdPartyMavenDepsListForBinary("//:test_oppia")
+    assertThat("@maven//:com_android_support_support_annotations")
+      .isNotIn(thirdPartyDependenciesList)
```

Comment:
Here & elsewhere: this is backwards. Truth is always assertThat(subject).hasOutcome(). I suggest changing this to:

``assertThat(thirdPartyDependenciesList).contains("@maven//:...")``

Ditto above.

Reply:
Done.